generate_fixtures.py

```diff
@@ -76,7 +76,17 @@ def write_and_import_keypair(self, name_dst):
         public_key = rt.import_ed25519_publickey_from_file(pathpub_src)
         private_key = rt.import_ed25519_privatekey_from_file(
             pathpriv_src, password='pw')
-        return (public_key, private_key)
+        return (public_key, private_key)\
+
+    def delegate_role(self, delegator_role, delegated_role_name, paths, file_name):
```

Comment:
I don't think this method should also be able to add a target; things that call this function should create the delegation, then add the target separately.

Reply:
This method is able to replace 7 instance where we create a delegation and immediately add a file with the delegation. 

The only place where call `.delegate()` and we can't replace it is in `TUFTestFixtureUnsupportedDelegation` because it is testing that `path_hash_prefixes` is **not** supported. 

I changed the name to `delegate_role_with_file`. I don't see a problem with having this all in 1 method especially since the named parameters make it very clear what the arguments are for.

This not to create an API that others can use but rather just to make our test more readable which I think it does.

Reply:
I updated parameter name from `file_name` to `target_file` to clearer

Reply:
> This not to create an API that others can use but rather just to make our test more readable which I think it does.

It's not an API for _others_ to use, that's true ...but it IS part of our internal testing API, much like BrowserTestBase is part of Drupal's. Therefore, I think it _should_ be treated as an important internal API, even if we don't expect others to use it, and we should not let it get sloppy. So I'm still in favor of making them separate calls. I don't really want to block it on this point, but to be honest, I do feel rather strongly about this.

Reply:
They python library we are using already provides `delegate_role()` as `tuf.repository_tool.Targets.delegate()`. `delegate_role_with_file()` is already calls this.

Reply:
Ok I created `delegate_role()`. It is only called 1 time, in `delegate_role_with_file()`.

I don't want to replace the call to `tuf.repository_tool.Targets.delegate()` in `TUFTestFixtureUnsupportedDelegation` because we would have to complicate `delegate_role()` with the extra `path_hash_prefixes` which we only need to prove it is not supported.

Because `delegate_role()` is only called 1x I would rather not create it now and just create it when we need it.  but I don't want to block the issue on it.